core/ledger/kvledger/txmgmt/privacyenabledstate/db.go

```diff
@@ -95,11 +105,25 @@ func (p *DBProvider) RegisterHealthChecker() error {
 }
 
 // GetDBHandle gets a handle to DB for a given id, i.e., a channel
-func (p *DBProvider) GetDBHandle(id string) (*DB, error) {
+func (p *DBProvider) GetDBHandle(id string, channelconfigProvider channelInfoProvider) (*DB, error) {
 	vdb, err := p.VersionedDBProvider.GetDBHandle(id)
 	if err != nil {
 		return nil, err
 	}
+
+	if db, ok := vdb.(statedb.ChannelMetadataCreator); ok {
+		if !db.ChannelMetadataExists() {
+			logger.Warnf("No channel metadata found for channel %s, creating it now...", id)
+			namespaces, err := constructStatedbNamespaces(id, vdb, channelconfigProvider, p.deployedccInfoProvider)
+			if err != nil {
+				return nil, err
+			}
+			if err = db.CreateChannelMetadata(namespaces); err != nil {
+				return nil, err
+			}
+		}
+	}
+
 	bookkeeper := p.bookkeepingProvider.GetDBHandle(id, bookkeeping.MetadataPresenceIndicator)
 	metadataHint := newMetadataHint(bookkeeper)
 	return NewDB(vdb, id, metadataHint)
```

Comment:
To reduce the extra code indentation. 

```suggestion
	bookkeeper := p.bookkeepingProvider.GetDBHandle(id, bookkeeping.MetadataPresenceIndicator)
	metadataHint := newMetadataHint(bookkeeper)
	
	db, ok := vdb.(statedb.ChannelMetadataCreator)
	if !ok || db.ChannelMetadataExists() {
	        	return NewDB(vdb, id, metadataHint)
	}
	logger.Warnf("No channel metadata found for channel %s, creating it now...", id)
	namespaces, err := constructStatedbNamespaces(id, vdb, channelconfigProvider, p.deployedccInfoProvider)
	if err != nil {
		return nil, err
	}
	if err = db.CreateChannelMetadata(namespaces); err != nil {
		return nil, err
	}
	return NewDB(vdb, id, metadataHint)
```